Fix: Correct fluentd prase TimeFormat config key

Jakob3xD opened the pull request:
<!--
Thank you for contributing to Fluent Operator!
Your commits need to follow DCO: https://probot.github.io/apps/dco/
And please provide the following information to help us make the most of your pull request:
-->

### What this PR does / why we need it:
It changes the TimeFormat config value from `time_type` to `time_format` as it should be.

### Which issue(s) this PR fixes:
Fixes #778
Fixes #781 

### Does this PR introduced a user-facing change?
<!--
If no, just write "None" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md
-->
```release-note
Fix: Correct fluentd prase TimeFormat config key
Fix: Correct Kubebuilder validation for TimeType from *string to string
```

### Additional documentation, usage docs, etc.:
<!--
This section can be blank if this pull request does not require a release note.
Please use the following format for linking documentation or pass the
section below:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

benjaminhuo commented on apis/fluentd/v1alpha1/plugins/common/common_types.go:
```diff
@@ -20,10 +20,10 @@ type CommonFields struct {
+	// +kubebuilder:validation:Enum:=float;unixtime;string;mixed
 	TimeType *string `json:"timeType,omitempty"`
-	// Process value according to the specified format. This is available only when time_type is *string
+	// Process value according to the specified format. This is available only when time_type is string
```
should change like this?
```suggestion
	// Process value according to the specified format. This is available only when time_format is string
```

wenchajun commented:
https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/helper_test.go#L355
It also seems to need revision here.

benjaminhuo commented:
> https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/helper_test.go#L355 It also seems to need revision here.

Yes, the CI error also needs to be fixed @Jakob3xD 
https://github.com/fluent/fluent-operator/blob/master/apis/fluentd/v1alpha1/tests/helper_test.go#L556
![image](https://github.com/fluent/fluent-operator/assets/18525465/9e2c3b78-a994-447f-845e-87f1430c9822)

Jakob3xD commented:
> Yes, the CI error also needs to be fixed @Jakob3xD

Should be fixed now. I updated the key and removed the new line at the end of file which was added by my vim.

benjaminhuo commented:
@Jakob3xD Thank you
